crates/runtime/src/libcalls.rs

```diff
@@ -121,28 +118,36 @@ pub mod trampolines {
                 // like s390x need to use outlined assembly files which requires
                 // `no_mangle`.
                 #[cfg_attr(target_arch = "s390x", wasmtime_versioned_export_macros::versioned_export)]
-                $( #[ $attr ] )*
+                #[allow(unused_variables)]
                 unsafe extern "C" fn [<impl_ $name>](
                     vmctx: *mut VMContext,
                     $( $pname : libcall!(@ty $param), )*
                 ) $( -> libcall!(@ty $result))? {
+                    $(#[cfg($attr)])?
+                    {
                     let result = std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| {
                         Instance::from_vmctx(vmctx, |instance| {
-                            super::$name(instance, $($pname),*)
+                            {
+                                super::$name(instance, $($pname),*)
+                            }
                         })
                     }));
                     match result {
                         Ok(ret) => LibcallResult::convert(ret),
                         Err(panic) => crate::traphandlers::resume_panic(panic),
                     }
+                    }
+                    $(
+                        #[cfg(not($attr))]
+                        std::process::abort();
+                    )?
```

Comment:
Why continue to define the function, but just have it abort? Is it really easier than not defining the function?